src/modules/ethereum/lockTransactions.js

```diff
@@ -120,16 +120,16 @@ export async function fetchLocksForAddressTx(lockManagerInstance, account) {
     return locks;
 }
 
-export async function processNewLockTx(account, lockData) {
+export async function processNewLockTx(account, event) {
     // reprdocue an array in same format as getLocksForAddress returnvalue:
     //     [ lockOwner, lockId, amountLocked, interestEarned, lockedUntil, perTermInterest, durationInSecs]
     const lockArray = [
-        lockData.lockId,
-        lockData.amountLocked,
-        lockData.interestEarned,
-        lockData.lockedUntil, // ethers.js passes these args to onnewlock as numbers unlike getLocksForAddress
-        lockData.perTermInterest,
-        lockData.durationInSecs,
+        event.returnValues.lockId,
```

Comment:
for both code and commit readability, it would be better to retain the original `lockData` variable name (just `const lockData = event.returnValues`)

Reply:
no guarantee to the order of attributes. returnValues has values redundantly under [0], [1] etc.  that might be used but bit more work than trivial.
can we leave it to an other PR?